tests/functional/cylc-cat-log/12-delete-kill.t

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env bash
+# THIS FILE IS PART OF THE CYLC WORKFLOW ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#-------------------------------------------------------------------------------
+# Test "cylc cat-log" exits when the log file is deleted
+# or when tail is killed.
+
+. "$(dirname "$0")/test_header"
+set_test_number 3
+
+init_workflow "${TEST_NAME_BASE}" << __EOF__
+# whatever
+__EOF__
+
+log_file="${WORKFLOW_RUN_DIR}/log/foo.log"
+echo "Hello, Mr. Thompson" > "$log_file"
+
+TEST_NAME="${TEST_NAME_BASE}-delete"
+cylc cat-log --mode=tail "$WORKFLOW_NAME" -f foo.log 2>&1 &
+cat_log_pid="$!"
+# Wait for tail to start
+sleep 3
```

Comment:
Sleeping can be fragile on loaded systems and increases test runtime, suggest polling for the fist line of the file, there's a test function for this:

```
cylc cat-log --mode=tail "$WORKFLOW_NAME" -f foo.log 2>&1 >cat-log.out &
cat_log_pid="$!"
# Wait for tail to start
poll_grep '<line to match>' cat-log.out
```